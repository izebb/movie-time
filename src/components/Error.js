import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  max-width: 500px;
  padding: 0 40px;
  margin: 100px auto 20px auto;
  text-align: center;
  font-size: 25px;
  font-weight: 600;
  color: #444;
`;

export function Error({ message, retry }) {
  const handleClick = () => {
    retry();
  };

  return (
    <Container>
      <p>{message}</p>
      <button onClick={handleClick}>Try again</button>
    </Container>
  );
}

Error.defaultProps = {
  retry: () => {},
  message: ''
};
