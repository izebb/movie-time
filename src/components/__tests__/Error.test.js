import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Error } from '../Error';

describe('Error', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Error />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
