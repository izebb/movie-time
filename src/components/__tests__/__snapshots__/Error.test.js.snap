// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render without crashing 1`] = `
<Error__Container>
  <p />
  <button
    onClick={[Function]}
  >
    Try again
  </button>
</Error__Container>
`;
